File: android/app/src/main/java/com/example/yugiohscanner/ml/CardLabels.kt
package com.example.yugiohscanner.ml

/**
 * Deutsche Anzeige der englischen Kartendaten (YGOPRODeck: "Spell Card", "DARK", "Dragon") in der Auswertung,
 * Feinschliff 26.09.2026. ZWILLING von desktop/src/utils/cardLabels.js, gemeinsame Fixture
 * docs/fixtures/valuation/card-labels.json. Unbekannte Werte bleiben, wie sie sind.
 */
object CardLabels {
    val ATTRIBUTES = mapOf(
        "DARK" to "Finsternis", "LIGHT" to "Licht", "EARTH" to "Erde", "WATER" to "Wasser", "FIRE" to "Feuer",
        "WIND" to "Wind", "DIVINE" to "Göttlich",
    )
    val RACES = mapOf(
        "Aqua" to "Aqua", "Beast" to "Ungeheuer", "Beast-Warrior" to "Ungeheuer-Krieger", "Creator-God" to "Schöpfergott",
        "Cyberse" to "Cyberse", "Dinosaur" to "Dinosaurier", "Divine-Beast" to "Göttliches Ungeheuer", "Dragon" to "Drache",
        "Fairy" to "Fee", "Fiend" to "Unterweltler", "Fish" to "Fisch", "Illusion" to "Illusion", "Insect" to "Insekt",
        "Machine" to "Maschine", "Plant" to "Pflanze", "Psychic" to "Psi", "Pyro" to "Pyro", "Reptile" to "Reptil",
        "Rock" to "Fels", "Sea Serpent" to "Seeschlange", "Spellcaster" to "Hexer", "Thunder" to "Donner",
        "Warrior" to "Krieger", "Winged Beast" to "Geflügeltes Ungeheuer", "Wyrm" to "Wyrm", "Zombie" to "Zombie",
        "Normal" to "Normal", "Continuous" to "Permanent", "Quick-Play" to "Schnell", "Field" to "Spielfeld",
        "Equip" to "Ausrüstung", "Ritual" to "Ritual", "Counter" to "Konter",
    )

    /** Kartenart als Gruppe: Zauber/Falle vor Monster geprüft (wie Dashboard.kt bisher). */
    fun typeGroup(type: String?): String = when {
        type?.contains("Spell", ignoreCase = true) == true -> "Zauber"
        type?.contains("Trap", ignoreCase = true) == true -> "Falle"
        type?.contains("Monster", ignoreCase = true) == true -> "Monster"
        else -> "Sonstige"
    }

    fun attribute(a: String): String = ATTRIBUTES[a] ?: a
    fun race(r: String): String = RACES[r] ?: r
}

File: desktop/src/utils/cardLabels.js
// Deutsche Anzeige der englischen Kartendaten (YGOPRODeck: "Spell Card", "DARK", "Dragon") in Statistik und Portfolio,
// Feinschliff 26.09.2026. ZWILLING: android .../ml/CardLabels.kt, gemeinsame Fixture docs/fixtures/valuation/card-labels.json.
// Unbekannte Werte bleiben, wie sie sind. Namen wie auf deutschen Konami-Karten.

export const ATTRIBUTE_LABELS = {
  DARK: 'Finsternis', LIGHT: 'Licht', EARTH: 'Erde', WATER: 'Wasser', FIRE: 'Feuer', WIND: 'Wind', DIVINE: 'Göttlich',
};

export const RACE_LABELS = {
  Aqua: 'Aqua', Beast: 'Ungeheuer', 'Beast-Warrior': 'Ungeheuer-Krieger', 'Creator-God': 'Schöpfergott', Cyberse: 'Cyberse',
  Dinosaur: 'Dinosaurier', 'Divine-Beast': 'Göttliches Ungeheuer', Dragon: 'Drache', Fairy: 'Fee', Fiend: 'Unterweltler',
  Fish: 'Fisch', Illusion: 'Illusion', Insect: 'Insekt', Machine: 'Maschine', Plant: 'Pflanze', Psychic: 'Psi',
  Pyro: 'Pyro', Reptile: 'Reptil', Rock: 'Fels', 'Sea Serpent': 'Seeschlange', Spellcaster: 'Hexer', Thunder: 'Donner',
  Warrior: 'Krieger', 'Winged Beast': 'Geflügeltes Ungeheuer', Wyrm: 'Wyrm', Zombie: 'Zombie',
  Normal: 'Normal', Continuous: 'Permanent', 'Quick-Play': 'Schnell', Field: 'Spielfeld', Equip: 'Ausrüstung',
  Ritual: 'Ritual', Counter: 'Konter',
};

// Kartenart als Gruppe (gleiche Reihenfolge der Prüfung wie Dashboard.kt#typeGroup: Zauber/Falle vor Monster).
export function typeGroup(type) {
  const t = String(type || '');
  if (/spell/i.test(t)) return 'Zauber';
  if (/trap/i.test(t)) return 'Falle';
  if (/monster/i.test(t)) return 'Monster';
  return 'Sonstige';
}

export const attributeLabel = (a) => ATTRIBUTE_LABELS[a] ?? a;
export const raceLabel = (r) => RACE_LABELS[r] ?? r;
